cfme/tests/configure/test_ntp_server.py

```diff
@@ -0,0 +1,89 @@
+import fauxfactory
+from cfme import login
+from cfme.configure import configuration
+from datetime import datetime
+from fixtures.pytest_store import store
+from utils.conf import cfme_data
+from utils.log import logger
+from utils.ssh import SSHClient
+from utils.wait import wait_for
+
+
+def ssh_command_execution(command):
+    with SSHClient() as ssh:
+        status, msg = ssh.run_command("%s" % command)
+        return status, msg
+
+
+def appliance_date():
+    status, msg = ssh_command_execution("date --iso-8601=hours")
+    return datetime.strptime(msg.rsplit('-', 1)[0], '%Y-%m-%dT%H')
+
+
+def test_ntp_crud():
+    """ Insert, Update and Delete the NTP servers """
+    # set from yaml file
+    configuration.configure_ntp_servers()
+
+    # set from random values
+    configuration.configure_ntp_servers(*(fauxfactory.gen_alphanumeric() for _ in range(3)))
+
+    configuration.unset_ntp_servers()
+
+
+def test_ntp_server_max_character():
+    configuration.configure_ntp_servers(*(fauxfactory.gen_alphanumeric(255) for _ in range(3)))
+
+
+def check_ntp_grep(clock):
+    status, msg = ssh_command_execution("cat /etc/ntp.conf | grep %s" % clock)
+    return True if status == 0 else False
+
+
+def test_ntp_conf_file_update_check():
+    # Adding the ntp server values
+    configuration.configure_ntp_servers()
+    for clock in cfme_data['clock_servers']:
```

Comment:
There is some flawed logic here. Your assertion should be inside the loop otherwise only the last ```wait_for``` needs to return ```True``` for the assertion not to blow up.